app/workers/builds.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+use Appwrite\Resque\Worker;
+use Cron\CronExpression;
+use Utopia\Database\Validator\Authorization;
+use Utopia\App;
+use Utopia\CLI\Console;
+use Utopia\Storage\Storage;
+use Utopia\Database\Document;
+use Utopia\Config\Config;
+
+require_once __DIR__.'/../init.php';
+
+Console::title('Builds V1 Worker');
+Console::success(APP_NAME.' build worker v1 has started');
+
+class BuildsV1 extends Worker
+{ 
+
+    public function getName(): string {
+        return "builds";
+    }
+
+    public function init(): void
+    {
+        Console::success("Initializing...");
+    }
+
+    public function run(): void
+    {
+        $type = $this->args['type'] ?? '';
+        $projectId = $this->args['projectId'] ?? '';
+
+        switch ($type) {
+            case BUILD_TYPE_TAG:
+                $functionId = $this->args['functionId'] ?? '';
+                $tagId = $this->args['tagId'] ?? '';
+                Console::success("Creating build for tag: $tagId");
+                $this->buildTag($projectId, $functionId, $tagId);
+                break;
+
+            case BUILD_TYPE_RETRY:
+                $buildId = $this->args['buildId'] ?? '';
+                Console::success("Retrying build for id: $buildId");
+                $this->triggerBuildStage($projectId, $buildId,);
+                break;
+
+            default:
+                throw new \Exception('Invalid build type');
+                break;
+        }
+    }
+
+    protected function triggerBuildStage(string $projectId, string $buildId)
+    {
+        // TODO: What is a reasonable time to wait for a build to complete?
```

Comment:
What should be a reasonable timeout for this ?

We're in the builds worker here. Does it matter if this request takes 900 seconds to complete? The rest of the builds will be queued in the meantime

Reply:
It really varies with builds from rust taking up to 20 minutes on slow hardware. Could we make it a environment variable?